src/json/any.cr

```diff
@@ -14,7 +14,8 @@
 # `Int32`, etc., use the `as_` methods, such as `#as_s`, `#as_i`, which perform
 # a type check against the raw underlying value. This means that invoking `#as_s`
 # when the underlying value is not a String will raise: the value won't automatically
-# be converted (parsed) to a `String`.
+# be converted (parsed) to a `String`.  For dynamic structures you can use
+# `#as_i?`, `as_h?` etc. to check structure at runtime.
```

Comment:
What does dynamic structures have to do with `to_i?`?

Reply:
You can use it to check if a node is an array vs. an int if it's "sometimes one or the other" or is "unknown" at compile time. Is there a better wording here?

Reply:
Is there a better wording you'd like here?